paper/paper.md

```diff
@@ -0,0 +1,57 @@
+---
+title: 'Streamz: A small library to manage continuous streams of data'
+tags:
+  - Stream
+  - Dask
+authors:
+ - name: Matthew Rocklin
+   orcid: ?
+   affiliation: 1
+ - name: Christopher J. Wright
+   orcid: 0000-0003-2522-7028
+   affiliation: 2
+ - name: Julien Lhermitte
+   orcid: 0000-0003-0660-975X
+   affiliation: 5
+ - name: Daniel B. Allan
+   orcid: 0000-0002-5947-6017
+   affiliation: 4
+ - name: Stuart I. Campbell
+   orcid: 0000-0001-7079-0878
+   affiliation: 4
+ - name: Kevin G. Yager
+   orcid: 0000-0001-7745-2513
+   affiliation: 5
+ - name: Simon J. L. Billinge
+   orcid: 0000-0002-9734-4998
+   affiliation: 2, 3
+affiliations:
+ - name: Anaconda Inc.
+   index: 1
+ - name: Department of Applied Physics and Applied Mathematics, Columbia University
+   index: 2
+ - name: Condensed Matter Physics and Materials Science Department, Brookhaven National Laboratory
+   index: 3
+ - name: NSLS-II, Brookhaven National Laboratory
+   index: 4
+ - name: Center for Functional Nanomaterials, Brookhaven National Laboratory
+   index: 5
+date: 19 September 2017
+bibliography: paper.bib
+---
+
+# Summary
+
+- This is a small library to manage continuous streams of data, particularly when complex branching and control flow situations arise. 
+- This provides a framework for streaming data analysis in pure Python with hooks into the Dask task scheduler for automatic parallelization.
+- Streamz is similar to reactive
+programming systems like RxPY [@RxPy] or big
+data streaming systems like Apache Flink [@Flink], Apache Beam [@Beam] or Apache Spark Streaming [@Spark].
+- This software forms the backbone for data processing at the NSLS-II X-ray powder diffraction and complex materials scattering data analysis pipelines.
```

Comment:
What about @mrocklin 's comment? I liked his suggestion. Copying it here again  for convenience:

> We describe the high-level architecture of the project, along with concrete real-world examples in data processing >pipelines in X-ray beamline facilities, particularly ....

Reply:
That's fair. I've been trying to read more JOSS papers to understand the proper length. Some of the papers seem very short, but others seem a bit longer (although all are under one page ish).